src/helpers/codeBlock.ts

```diff
@@ -1,43 +1,75 @@
-import { lexer as markdown } from "marked";
-
+import { parse } from "./contentLexer";
 import { createBin } from "./createBin";
 
-export async function blockMatcher(content: string): Promise<Map<string, { language?: string; code: string }>> {
-	const matches = markdown(content);
-	const blocks = new Map<string, { language?: string; code: string }>();
-
-	if (!matches) {
-		return blocks;
-	}
-
-	const MAX_LINES = parseInt(process.env.MAX_LINES!, 10);
+const MAX_LINES = parseInt(process.env.MAX_LINES!, 10);
 
-	for (const block of matches) {
-		if (!("lang" in block)) {
-			continue;
-		}
-
-		const { lang: language, text: code } = block;
-		const raw = block.raw.trim();
+interface TextNode {
+	type: "text";
+	content: string;
+}
+interface CodeNode {
+	type: "code";
+	content: string;
+	lang: string;
+	raw: string;
+	inline: boolean;
+	isBig?: boolean;
+}
+interface BlankNode {
+	type: "blank";
+}
+type Parsed = Array<CodeNode | TextNode | BlankNode>;
 
-		if (code.split("\n", MAX_LINES).length === MAX_LINES && !blocks.has(raw)) {
-			blocks.set(raw, { language, code });
+export function blockMatcher(content: string): Parsed {
+	try {
+		const matches: Parsed = parse(content);
+		let containsBigCode = false;
+		for (let i = 0; i < matches.length; i++) {
```

Comment:
Je sais pas si un `for (const match in matches) {}` (ici du coup match correspond à `i`) serait pas mieux

Reply:
C'est ce que j'ai fais mais eslint m'a gueuler dessus, et j'avais pas envie de me casser la tête